### PR TITLE
parse esdt transfers

### PR DESCRIPTION
**Pull request by sasurobert**
parse esdt transfers

### CODE REVIEW

File: parsers/esdtTransferParser.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+// ParseESDTTransfers returns the list of esdt transfers, the callFunction and callArgs from the given arguments
+func (e *esdtTransferParser) ParseESDTTransfers(
+	sndAddr []byte,
+	rcvAddr []byte,
+	function string,
+	args [][]byte,
+) (*vmcommon.ParsedESDTTransfers, error) {
+	switch function {
```
Comment (iulianpascalau) [inline parsers/esdtTransferParser.go:56]: we can avoid this switch by further splitting this implementation in smaller, dedicated sub-parsers.

File: parsers/esdtTransferParser.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		esdtTransfers.CallFunction = string(args[minLenArgs])
+	}
+	if uint64(len(args)) > minLenArgs+1 {
+		esdtTransfers.CallArgs = append(esdtTransfers.CallArgs, args[minLenArgs+1:]...)
+	}
+
+	esdtTransfers.ESDTTransfers = make([]*vmcommon.ESDTTransfer, numOfTransfer.Uint64())
+	for i := uint64(0); i < numOfTransfer.Uint64(); i++ {
+		tokenStartIndex := startIndex + i*ArgsPerTransfer
+		esdtTransfers.ESDTTransfers[i] = &vmcommon.ESDTTransfer{
```
Comment (iulianpascalau) [inline parsers/esdtTransferParser.go:156]: the creation of the new ESDTTransfer instance along with the following if could have been extracted in a new function

**Review by SebastianMarian — DISMISSED** [review]

File: parsers/esdtTransferParser.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	"bytes"
+	"errors"
+	"math/big"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/ElrondNetwork/elrond-vm-common/check"
+	"github.com/ElrondNetwork/elrond-vm-common/data/esdt"
+)
+
+// ErrNotESDTTransferInput signals invalid ESDT transfer input error
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply parsers/esdtTransferParser.go:13]: done

File: parsers/esdtTransferParser.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+var ErrNotESDTTransferInput = errors.New("not ESDT transfer input")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply parsers/esdtTransferParser.go:14]: done

File: parsers/esdtTransferParser.go (hunk 1 above)
Comment (sasurobert) [reply parsers/esdtTransferParser.go:56]: yeah and a container for that and a get from container. I would leave like this for now.

File: parsers/esdtTransferParser.go (hunk 2 above)
Comment (sasurobert) [reply parsers/esdtTransferParser.go:159]: moved

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**